### PR TITLE
Moved Test URLs from hardcoded to environment variables

### LINKED ISSUE #94: Configure tests to use environment variables for goerli urls, rather than hardcoded urls
Providers seem to be initialized from a hardcoded url as shown in places like [here](https://github.com/a16z/magi/blob/98bd1abefa91009af9f8b1a708ea09a4f146044b/src/derive/mod.rs#L126). It would be better to refactor this load from a URL such as "GOERLI_RPC," so that it is easy to use a different provider, or even a local one, for testing. I think this approach also leaks an API key 🤔 

Would you be ok with me tackling this @ncitron or @refcell ? I don't have perms to assign myself

### PR DESCRIPTION
**Pull request by ControlCplusControlV**
Closes #94 

Hopefully my first PR of many, sorry for taking so long, been busy with school :)

Can't run CI, I think there may be an issue at first though with setting the new env variables, but once that is fixed it all should be good

### CODE REVIEW

**Comment by refcell** [comment]
Note: This will require @ncitron to set the env vars in the github repo as well as exposing them to github actions through the github workflow ci config.

File: docker/.env.default [hunk 1]
```diff
@@ -4,6 +4,12 @@ NETWORK=optimism-goerli
 # L1 RPC URL
 L1_RPC_URL=https://eth-goerli.g.alchemy.com/v2/<API_KEY>
 
+# L1 Test RPC URL
+L1_TEST_RPC_URL=https://eth-goerli.g.alchemy.com/v2/a--NIcyeycPntQX42kunxUIVkg6_ekYc
```
Comment (ncitron) [inline docker/.env.default:8]: I don't think we need to include these URLs in the docker configs. The docker setup is not meant to be used to run the tests, but rather for people to run magi as a whole.

Another good reason I want to remove them here is to make sure people aren't tempted to use my keys to sync their node, which will eat up my rate limit quite quickly.

**Review by ncitron — COMMENTED** [review]
LGTM! Just one nit on removing the urls from the docker envs.

I'll also add these keys to our github secrets.

File: docker/.env.default (hunk 1 above)
Comment (ControlCplusControlV) [reply docker/.env.default:8]: Good point, I was surprised you were willing to just leave them in the code, but I guess this makes it even easier for scrapers to use them. Will remove them

**Review by ncitron — APPROVED** [review]
LGTM

**Pull request merged**